actors/miner/src/expiration_queue.rs

```diff
@@ -877,27 +953,33 @@ fn group_new_sectors_by_declared_expiration<'a>(
     let mut sectors_by_expiration = BTreeMap::<ChainEpoch, Vec<&SectorOnChainInfo>>::new();
 
     for sector in sectors {
-        let q_expiration = quant.quantize_up(sector.expiration);
+        let q_expiration = quant.quantize_up(sector.expires_at());
         sectors_by_expiration.entry(q_expiration).or_default().push(sector);
     }
 
     // The result is sorted by expiration because the BTreeMap iterates in sorted order.
     sectors_by_expiration
         .into_iter()
         .map(|(expiration, epoch_sectors)| {
-            let mut sector_numbers = Vec::<u64>::with_capacity(epoch_sectors.len());
+            let mut sectors_on_time = Vec::new();
+            let mut sectors_early = Vec::new();
             let mut total_power = PowerPair::zero();
             let mut total_pledge = TokenAmount::zero();
 
             for sector in epoch_sectors {
-                sector_numbers.push(sector.sector_number);
                 total_power += &power_for_sector(sector_size, sector);
-                total_pledge += &sector.initial_pledge;
+                if sector.expires_early() {
+                    sectors_early.push(sector.sector_number);
+                } else {
+                    total_pledge += &sector.initial_pledge;
```

Comment:
Why is this only added in the not-early case? If this is intentional, please add documentation to the SectorEpochSet fields so that this makes sense.

Reply:
Yes it is intentional, as the early expirations function via the fault expiration mechanism.

Reply:
And this is only used for releasing pledge when they sector expires.